### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by stephnna**
Hi code reviewer,

I this project, I implemented the following

Splash screen

A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
Sign up and log in pages

The user should be able to register in the app with full name, email and password (all mandatory).
The user can log into the app using email and password.
If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
Home page (catagories page)

When the user logs in, they are presented with the categories page.
For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
Transactions page

For a given category, the list of transactions is presented, ordered by the most recent.
At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new category" page

The user fills out a form to create a new category, indicating their name and icon (both mandatory).
The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new transaction" page

The user fills out a form to create a new transaction with:
name (mandatory)
amount (mandatory)
categories (mandatory at least one)
The user click (or taps) the "Save" button to create the n
ew transation, and is taken to the transactions page for that category.
When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
Testing requirements
Create unit and integration tests for all the most important components of your RoR application.
Technical requirements
You should use Postgres as your database.
You should use devise for authentication.
You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
You can use a view template engine of your choice (.erb, .slim, .haml).
The project should be deployed and accessible online.
Kindly review as required, 
thanks

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,218 @@
...
+- where you can manage your budget: you have a list of transactions associated with a category, so that you can see how much money you spent and on what.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+- [Video Demo Link](https://www.loom.com/share/ca59940837a04b618b95d787c663d174)
+
+- [Live Demo Link]()
```
Comment (ShadyShawkat) [inline README.md:94]: - [ ] Kindly deploy your app and add the deployment link in the readme file, as per [this](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=is%20tested%20adequately.-,Deploy%20the%20project%2C,-and%20test%20for) requirement.

File: README.md [hunk 2]
```diff
@@ -1 +1,218 @@
...
+If you like this project. Don't forget to give it a ⭐️
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+- A big thanks to [@microverseinc](https://github.com/microverseinc) for the project idea, guidance and inspiration.
+
```
Comment (ShadyShawkat) [inline README.md:199]: - [ ] Please make sure to give appropriate credit where it's due to the original creator of the design in the `Acknowledgments` section in your readme file, as per [this](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=The%20Creative%20Commons%20license%20of%20the%20design%20requires%20that%20you%20give%20appropriate%20credit%20to%20the%20author.%20Therefore%2C%20you%20must%20do%20it%20in%20the%20README%20of%20your%20project.) requirement.

File: spec/models/category_spec.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe Category, type: :model do
+  subject do
+    @user = User.new(name: 'Stephen', email: '123@mail.com', password: 'password')
+    @group = @user.categories.new(name: 'Travel', icon: 'https://pxhere.com/en/photo/1569711', user_id: @user.id)
+  end
+  before { subject.save }
+
```
Comment (ShadyShawkat) [inline spec/models/category_spec.rb:9]: - [ ] Good job adding unit tests for the models. Please also add integration tests for the app components.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @stephnna, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (stephnna) [reply README.md:199]: Done!

File: README.md (hunk 1 above)
Comment (stephnna) [reply README.md:94]: Hi @ShadyShawkat 


Thanks for that observation, Micoverse asked us to deploy to Heroku but unfortunately, Heroku does not support free web hosting anymore! This began on Nov 28 this year. For now, the most important thing is that my app is working as you can see from my loom video. On my own, I will search for another free rail web hosting site, learn how to deploy using their platform, and deploy it later. Kindly approve the project.
Thank you!

File: spec/models/category_spec.rb (hunk 3 above)
Comment (stephnna) [reply spec/models/category_spec.rb:9]: Done
Integration test added
Thanks!

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @stephnna 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Well done making the requested changes
- Great communication

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**